### ISSUE TITLE
Trigger group can be left as paused when all triggers have been removed

### ISSUE THREAD

**Issue by MPapst** [quartznet/quartznet#641]
### Version: x.y.z
3.0.5

### Expected behavior

After removing the first Job, the second one should be executed when scheduled.

### Actual behavior

After deleting the first job, the second one is never triggered.

Edit:  
*This only happens when deleting all Jobs*. When there is one Job left, it works. Unfortunaltey I am unable to edit the title.

#### Output from sample below
```
Called 05.07.2018 08:44:44 +00:00
Called 05.07.2018 08:44:46 +00:00
Called 05.07.2018 08:44:48 +00:00
Called 05.07.2018 08:44:50 +00:00
Called 05.07.2018 08:44:52 +00:00
Called 05.07.2018 08:44:54 +00:00
```

Missing expected output from Class `Job2`.

### Steps to reproduce

```csharp
async Task Main()
{
	IScheduler sched = await new StdSchedulerFactory().GetScheduler();
	
	var job = JobBuilder.Create<Job>().WithIdentity("test").Build();
	var trigger = TriggerBuilder.Create().StartNow().WithSimpleSchedule(x => x.RepeatForever().WithIntervalInSeconds(2)).Build();
	
	await sched.Start();
	
	await sched.ScheduleJob(job, trigger);
	
	await Task.Delay(TimeSpan.FromSeconds(10));
	
	await sched.PauseAll();
	await sched.DeleteJob(job.Key);
	await sched.ResumeAll();
	
	await Task.Delay(TimeSpan.FromSeconds(1));
	
	var job2 = JobBuilder.Create<Job2>().WithIdentity("test2").Build();
	var trigger2 = TriggerBuilder.Create().StartNow().WithSimpleSchedule(x => x.RepeatForever().WithIntervalInSeconds(1)).Build();
	
	
	await sched.Start();
	
	await sched.ScheduleJob(job2, trigger2);
	
	await Task.Delay(TimeSpan.FromSeconds(10));
	
	await sched.Shutdown();
}

// Define other methods and classes here
class Job : IJob
{
	public Task Execute(IJobExecutionContext context)
	{
		Console.WriteLine($"Called {context.FireTimeUtc}");
		return Task.CompletedTask;
	}
}

class Job2 : IJob
{
	public Task Execute(IJobExecutionContext context)
	{
		Console.WriteLine($"Called 2 {context.FireTimeUtc}");
		return Task.CompletedTask;
	}
}
```
Note: Sample is run in LINQPad 5.

**Comment by cedral** [comment]
What are you calling sched.Start() twice? It looks the schedule is already started and then you try to start it again. (my experience stops with version 2.6 so if I am off base I apologize.)

**Comment by MPapst** [comment]
That was just a try :) But did not have any effect.
I am a bit further right now:

* it does not make any difference pausing and resuming jobs, behavior is the same
* it does make difference calling sched.Clear() instead of deleting the last Job, the scheduler is then running and triggering `Job2`

**Comment by lahma** [comment]
Thanks for reporting this, should be fixed now and part of 3.0.6 release.
